src/library/scala/Array.scala

```diff
@@ -106,7 +106,9 @@ object Array {
    */
   def copy(src: AnyRef, srcPos: Int, dest: AnyRef, destPos: Int, length: Int): Unit = {
     val srcClass = src.getClass
-    if (srcClass.isArray && dest.getClass.isAssignableFrom(srcClass))
+    val destClass = dest.getClass
+    if (srcClass.isArray && ((destClass eq srcClass) ||
```

Comment:
IIUC there's always only one `Class` instance for a given class (or `Array[C]`) - is that documented?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html

> Every array also belongs to a class that is reflected as a Class object that is _shared by all arrays_ with the same element type and number of dimensions.

(emphasis mine)